packages/common/pools/src/db/sqlite/mod.rs

```diff
@@ -620,7 +678,40 @@ impl SqlitePoolManager {
 						chunk_count += 1;
 
 						// Write to buffer
-						buf.extend(entry.value());
+						compressed_data_buf.extend(entry.value());
+					}
+
+					// Decompress the data
+					let mut decoder = lz4_flex::frame::FrameDecoder::new(&compressed_data_buf[..]);
+					decoder
+						.read_to_end(&mut buf)
+						.map_err(Error::Io)
+						.map_err(|x| fdb::FdbBindingError::CustomError(x.into()))?;
+
+					// If there is no compressed data, read from the uncompressed data (backwards compatibility)
+					if chunk_count == 0 {
+						while let Some(entry) = data_stream.try_next().await? {
```

Comment:
we can move reading the data stream in to here. this reduces the overall reads at the cost of higher latency for legacy wfs.